kernel/arch/dreamcast/kernel/timer.c

```diff
@@ -152,25 +152,36 @@ int timer_clear(int which) {
     TIMER16(tcrs[which]) &= ~UNF;
     return !!(value & UNF);
 }
-
 /* Spin-loop kernel sleep func: uses the secondary timer in the
    SH-4 to very accurately delay even when interrupts are disabled */
-void timer_spin_sleep(int ms) {
-    timer_prime(TMU1, 1000, 0);
+static void timer_spin_sleep_ticks(unsigned speed, int ticks) {
+    timer_prime(TMU1, speed, 0);
     timer_clear(TMU1);
     timer_start(TMU1);
 
-    while(ms > 0) {
+    while(ticks > 0) {
         while(!(TIMER16(tcrs[TMU1]) & UNF))
```

Comment:
Would it make sense to have this be based around a generic version of the newer timer_getticks that takes in the timer # as a param?

Reply:
yeah, that might actually be a good idea.